integration/test/performancetest/performance_test.go

```diff
@@ -1,59 +1,246 @@
-// go:build linux && integration
+//go:build linux && integration
 // +build linux,integration
 
 package performancetest
 
-import(
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	//"strconv"
+	"sync"
 	"testing"
 	"time"
-	"log"
-	"context"
+
 	"github.com/aws/amazon-cloudwatch-agent/integration/test"
 )
 
 const (
-	configPath = "resources/config.json"
 	configOutputPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
-	agentRuntimeMinutes = 5 //20min desired but 5mins for testing purposes 
+	agentRuntimeMinutes = 5 //20 mins desired but 5 mins for testing purposes
 	DynamoDBDataBase = "CWAPerformanceMetrics"
-
+	testLogNum = "PERFORMANCE_NUMBER_OF_LOGS"
 )
 
+//this struct is derived from plugins/inputs/logfile FileConfig struct
+type LogInfo struct {
+	FilePath       string `json:"file_path"`
+	LogGroupName   string `json:"log_group_name"`
+	LogStreamName  string `json:"log_stream_name"`
+	Timezone       string `json:"timezone"`
+}
+
 func TestPerformance(t *testing.T) {
+	//get number of logs for test from github action
+	//@TODO
+	//logNum, err := strconv.Atoi(os.Getenv(testLogNum)) //requires a commit from Okan that updates the workflow file so the log tests will run concurrently
+	// if err != nil {
+	// 	t.Fatalf("Error: cannot convert test log number to integer, %v", err)
+	// }
+	logNum := 10 //THIS IS TEMPORARY SO CODE RUNS
+
 	agentContext := context.TODO()
 	instanceId := test.GetInstanceId()
 	log.Printf("Instance ID used for performance metrics : %s\n", instanceId)
 
-	test.CopyFile(configPath, configOutputPath)
-
-	test.StartAgent(configOutputPath, true)
-
-	agentRunDuration := agentRuntimeMinutes * time.Minute
-	//let agent run before collecting performance metrics on it
-	time.Sleep(agentRunDuration)
-	log.Printf("Agent has been running for : %s\n", (agentRunDuration).String())
-	test.StopAgent()
-
-	//collect data
-	data, err := GetPerformanceMetrics(instanceId, agentRuntimeMinutes, agentContext)
-	//@TODO check if metrics are zero remove them and make sure there are non-zero metrics existing
+	configFilePath, logStreams, err := GenerateConfig(logNum)
 	if err != nil {
-		log.Println("Error: ", err)
 		t.Fatalf("Error: %v", err)
 	}
 
-	//------Placeholder to put data into database------//
-	//useless code so data get used and compiler isn't mad
-	if data == nil {
-		t.Fatalf("No data")
+	//defer deleting log group and streams
+	for _, logStream := range logStreams {
+		defer test.DeleteLogGroupAndStream(instanceId, logStream)
+	}
```

Comment:
I think this is fine, but an enhancement to this is to reduce the number of bad calls. The logs are being published to the same log group but different log streams. If we have 1000 log streams, in this we are making 1000 calls to delete a single log group. I would consider either modifying the function to take in a variable number of log stream names:
```go
func foo(group string, ...streams string) 
```
I think. 

Or, maybe we just want to separate it out so there is a function for deleting log groups, a function for deleting log streams, and then we gut the `DeleteLogGroupAndStream` to call both of those functions, and then we call them separately like:
```go
test.DeleteLogGroup(foo)
for _, logStream := range logStreams {
  test.DeleteLogStream(logStream)
}
```

Reply:
Just to clarify, are you proposing that I modify functions in or append functions to the cwl_util.go file?